### PR TITLE
APPT-186 - run e2e tests in pipeline

### PR DESCRIPTION
**Pull request by kicr1**
- Create a remote cosmos account for each pipeline build
- Remove cosmos account for canceled and successful builds
- Run integration tests in pipeline using containerised api and remote cosmos account
- Run E2E playwright tests in pipeline using containerised nextapp and other required services connected to remote cosmos account
- Output screenshots on failed E2E tests
- Generate and output azure devops test reports for integration and E2E tests
- Allow for bringing up docker containers locally using cosmos docker container
- Cosmos docker container not used in pipeline

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,11 +1,50 @@
...
+    profiles: [local]
     image: mcr.microsoft.com/cosmosdb/linux/azure-cosmos-emulator:latest
     ports:
       - "8081:8081"
-      - "10250-10255:10250-10255"
+
+  next-app:
+    container_name: next-app
+    build:
+      context: .
```
Comment (vicr1) [inline docker-compose.yml:11]: Could the context for this be src/new-client - might speed up the docker build a little bit

File: docker-compose.yml [hunk 2]
```diff
@@ -1,11 +1,50 @@
...
+      - ./src/new-client/src:/app/src
+      - ./src/new-client/public:/app/public
+    restart: always
+    ports:
+      - 3000:3000
+  
+  api:
+    container_name: api
+    build:
+      context: .
```
Comment (vicr1) [inline docker-compose.yml:28]: Could the context for this be src/api?

**Review by vicr1 — DISMISSED** [review]

**Review by vicr1 — DISMISSED** [review]

File: .vscode/tasks.json [hunk 3]
```diff
@@ -8,9 +8,10 @@
       "type": "docker-compose",
       "dockerCompose": {
         "up": {
+          "profile": "local",
           "detached": true,
           "build": true,
-          "services": ["cosmos", "mock-api", "oidc-server"]
+          "services": ["cosmos", "mock-api", "oidc-server", "next-app", "api"]
```
Comment (vicr1) [inline .vscode/tasks.json:14]: Do you need to specify services if you have profile? Could you omit one of these as they are both ways of specifying which services to bring up

**Review by vicr1 — APPROVED** [review]

File: .vscode/tasks.json (hunk 3 above)
Comment (kicr1) [reply .vscode/tasks.json]: Removed services now we are using the profile.

**Review by vicr1 — APPROVED** [review]

**Pull request merged**